lib/concurrent/atomic/atomic_fixnum.rb

```diff
@@ -74,10 +68,7 @@ def value=(value)
     #
     #   @return [Fixnum] the current value after incrementation
     def increment
-      @mutex.lock
-      @value += 1
-    ensure
-      @mutex.unlock
+      synchronize { @value += 1 }
```

Comment:
I know Java's int is max'd at 2^31 - 1, so it might make since to cap it at that for compatibility reasons.

On the contrary, if we do that we are re-interpreting what a Fixnum is, since we no longer roll over into BigNum automatically.